### ISSUE TITLE
BeginTx() panics when called on a DebugDriver

### ISSUE THREAD

**Issue by BrentChesny** [facebook/ent#1175]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Calling `BeginTx(ctx, opts)` on a debug Client panics, even though the underlying driver is an sql.Driver. The culprit seems to be this line in the BeginTx method of `*Client`:
```
tx, err := c.driver.(*sql.Driver).BeginTx(ctx, opts)
```

The type cast seems to fail since `c.driver` is of type `driver.DebugDriver`, even though it embeds an `*sql.Driver`.

## Expected Behavior 🤔

I should be able to call BeginTx on a debug driver without issues.

## Steps to Reproduce 🕹

Make the following call in your code:
```
client.Debug().BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
```
where `client` is an Ent client with an underlying sql driver.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.15.3  |
| Ent         | 0.5.4   |
| Database    | postgresql   |
| Driver      | https://github.com/lib/pq        |

**Comment by a8m** [comment]
Thanks for reporting this @BrentChesny. https://github.com/facebook/ent/pull/1176 should fix the issue.

**Comment by a8m** [comment]
Closed with #1176

**Comment by BrentChesny** [comment]
Thanks for the quick response as always. 🏆
